jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -207,6 +207,25 @@ public Type getBaseType()
         })
         .build();
 
+    public static Type getBaseType(String contentType)
+    {
+        if (StringUtil.isEmpty(contentType))
+            return null;
+        Type type = CACHE.getBest(contentType);
```

Comment:
What happens if the mime-type is a substring of another mime-type?

Eg: `application/xml` vs `application/xml-dtd` ?
or `video/h264` and `video/h264-rcdo` and `video/h264-svc`

Reply:
Note this is only for our known types in the enum. But also the code below. We check for how it terminates.
I'll add some more explicit tests for that.